### PR TITLE
refactor build setup for osx support

### PR DESCRIPTION
**Pull request by jeroenooms**
Tested on OSX 10.10 and Ubuntu 14.04.

### CODE REVIEW

File: src/Makevars.in [hunk 1]
```diff
@@ -29,6 +29,8 @@ BBG_RPATH   = @rpath@
 
 ## set include and linker options
 ## Bbg API files are assummed to be in the standard search path
-PKG_CPPFLAGS = -I../inst/include/ -I.
```
Comment (eddelbuettel) [inline src/Makevars.in]: I kinda prefer the headers in inst/ so that they get installed.

File: configure [hunk 2]
```diff
@@ -20,47 +21,52 @@
...
-    tar xfz blpLibrary.tar.gz libblpapi3_${flavour}.so
-    rm blpLibrary.tar.gz
-    
-    cd ../../inst
-    wget https://github.com/Rblp/blp/raw/master/headers/linux/blpHeaders.tar.gz
-    tar xfz blpHeaders.tar.gz
-    rm blpHeaders.tar.gz
-    
+# Populate Makevars
+rpath=$(Rscript -e 'cat(file.path(.libPaths()[1], "Rblpapi", "libs"))')
```
Comment (eddelbuettel) [inline configure]: I prefer `Rblpapi/blp` which has worked for us on Linux for a long time.

File: configure [hunk 3]
```diff
@@ -20,47 +21,52 @@
...
 
+# Get header files
+cd src
+curl -OL https://github.com/Rblp/blp/raw/master/headers/${platform}/blpHeaders.tar.gz
+tar xfz blpHeaders.tar.gz
+rm blpHeaders.tar.gz
+
+# Download precompiled shared library
+cd ..
+mkdir -p inst/libs
```
Comment (eddelbuettel) [inline configure]: I prefer `inst/blp` which has worked for us on Linux for a long time.

File: configure [hunk 4]
```diff
@@ -20,47 +21,52 @@
...
+    sed -e"s/@config@/blpapi3_32/" -e"s|@rpath@|"${rpath}"|" src/Makevars.in > src/Makevars
+    flavour="32"
 else
-    echo "OS X is not yet supported. Contributions are welcome."
+    echo "Unknown architecture: ${arch}. Exiting."
     exit -1
 fi
 
+# Get header files
+cd src
```
Comment (eddelbuettel) [inline configure]: I prefer `inst/include`.

File: src/Makevars.in (hunk 1 above)
Comment (jeroenooms) [inline src/Makevars.in]: Why do you need the headers installed? They have no use at runtime? Some distributions don't even allow headers and bin's to be shipped in the same package.

Comment (eddelbuettel) [inline src/Makevars.in]: I do not need them, but like them present. It is additional information.

**Comment by eddelbuettel** [comment]
Thanks for reverting `lib/` to `blp/`.

**Comment by eddelbuettel** [comment]
This morphed into #61

**Pull request closed without merging**